File: CS/E3428.Web/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace E3428.Web {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void XtraReport1_DataSourceDemanded(object sender, EventArgs e) {
            custOrdersDetailTableAdapter.Fill((this.DataSource as NwindDataSet).CustOrdersDetail, Convert.ToInt32(Parameters["OrderId"].Value));
        }

    }
}

File: VB/E3428.Web/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace E3428.Web
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub XtraReport1_DataSourceDemanded(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.DataSourceDemanded
			custOrdersDetailTableAdapter.Fill((TryCast(Me.DataSource, NwindDataSet)).CustOrdersDetail, Convert.ToInt32(Parameters("OrderId").Value))
		End Sub

	End Class
End Namespace
